### DISCUSSION TITLE
Only saying 'closes #issue' in PR is not enough

### DISCUSSION THREAD

**Discussion by ShooterIT** [apache/kvrocks#1179] [General]
I found recent PRs only say 'closes #xxx', i don't think it is enough, the issue maybe only describe `why` we should do things or there is a bug. But in the PR, we need to describe not only the problem but also the solution, even the pros and cons, and we should add these informations in comment log.

**Comment by tisonkun** [comment]
@ShooterIT If you find a specific PR without sufficient information/description, you as a committer can send a request changes review.

For some trivial changes like fix a typo or finish a subtasks whose tracking issue is clear defined (migrating tests, fixing clang-clippy linter), it needs not redundant description.

The GitHub-provided PULL_REQUEST_TEMPLATE mechanism can always grow too cumbersome to follow. I don't see a good practice for it - finally, most contributors omit it. (If it can support form like issue template, the case may be better, but not today.)

**Reply by git-hulk** [reply]
Yes, it's hard to require contributors to do that. As a committer, what we can do is ask for more information if the issue/PR is not clear enough, then the merger can help to summarize or rewrite the commit message first.

**Reply by tisonkun** [reply]
Also, review after merge can help to spread consensus. If a PR goes through a wrong process in your opinion, commenting on it under the thread is encouraged.
